synfig-studio/src/gui/widgets/widget_soundwave.cpp

```diff
@@ -197,7 +197,7 @@ bool Widget_SoundWave::on_draw(const Cairo::RefPtr<Cairo::Context>& cr)
 		int value = middle_y;
 		if (t >= sound_delay) {
 			synfig::Time dt = t - sound_delay;
-			int index = int(dt * stride) + channel_idx;
+			unsigned long index = int(dt * stride) + channel_idx;
 			if (index >= 0 && index < buffer.size())
```

Comment:
Just remove `index >= 0` since unsigned long can never be negative. We're unnecessarily comparing an always +ve value with +ve

Helpful post 👇
https://stackoverflow.com/questions/5066362/why-is-comparing-if-an-unsigned-int-0-a-pointless-comparison

Reply:
I am not sure this should be removed. Maybe it can get negative values, so we need to dig deeper here.

Reply:
The post said if a -ve is returned, `index` will just be defined with the largest unsigned number instead. However, deeper investigation could be done on what each variable of this statement `int(dt * stride) + channel_idx` does.

Reply:
Did a little investigation:
- `dt = t - sound_delay`. --> I think it will never be negative since it is a diff.
- `stride = frequency * bytes_per_sample * n_channels` --> I think it will also never be negative.
- `channel_idx`--> I am not sure about this, but I think this will never be negative. 

That being said `index = int(dt * stride) + channel_idx` will always positive.
I will dig deeper to make sure.